api/v1alpha1/hookaction_types.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:resource:path=hookaction,shortName=ha,scope=Namespaced,categories=hookaction
+// +kubebuilder:object:root=true
+
+// HookAction is the Schema for the hookactions API
```

Comment:
Add the comments from the KEP:

// HookAction describes action commands to run on pods/containers based
// on specified policies. HookAction will be created by the user and
// can be re-used later. Application Snapshot Controller will create ExecutionHooks
// based on HookActions specified in the snapshot spec. For example,
// two HookActions, preSnapshotExecutionHook and postSnapshotExecutionHook,
// are expected in the application snapshot spec.
// HookAction does not contain information on pods/containers because those are
// runtime info.